control/src/main/java/org/kaazing/k3po/control/internal/Control.java

```diff
@@ -230,6 +242,28 @@ private void writeCommand(AbortCommand abort) throws IOException, CharacterCodin
         textOut.flush();
     }
 
+    private void writeCommand(NotifyCommand notify) throws IOException, CharacterCodingException {
+        OutputStream bytesOut = connection.getOutputStream();
+        CharsetEncoder encoder = UTF_8.newEncoder();
+        Writer textOut = new OutputStreamWriter(bytesOut, encoder);
+
+        textOut.append("NOTIFY\n");
+        textOut.append(format("barrier:%s\n", notify.getBarrier()));
```

Comment:
I haven't seen the control protocol, wouldn't it be better if it looks like "NOTIFY barrier:foo"